src/answerer.rs

```diff
@@ -0,0 +1,355 @@
+                    .context("could not get new connection from pool")
+                {
+                    Ok(db_connection) => db_connection,
+                    Err(error) => {
+                        tracing::error!(
+                    "could not get database connection while trying to update oracle's answer: {:#}",
+                    error
+                );
+                        return Ok(());
+                    }
+                };
+
+                if let Err(error) = active_oracle.update_answer(&mut db_connection, answer) {
+                    tracing::error!("{:#}", error);
+                    return Ok(());
+                }
+            }
+            answer
+        }
+    };
+    if let Some(answer) = answer {
+        // if we arrive here, an answer is available and we should submit it
+
+        tracing::info!("answering with value {}", answer);
+        let oracle = DefiLlamaOracle::new(active_oracle.address.0, signer.clone());
+        let mut call = oracle.finalize(answer);
+
+        match signer.fill_transaction(&mut call.tx, None).await {
+            Ok(()) => {}
+            Err(error) => {
+                tracing::error!("could not fill answer call: {:#}", error);
+                return Ok(());
+            }
+        };
+
+        let tx = match call.send().await {
+            Ok(tx) => tx,
+            Err(error) => {
+                tracing::error!(
+                    "error while submitting answer transaction {:?}: {:#}",
+                    call.tx,
+                    error
+                );
+                return Ok(());
+            }
+        };
+
+        {
+            let mut db_connection = match db_connection_pool
+                .get()
+                .context("could not get new connection from pool")
+            {
+                Ok(db_connection) => db_connection,
+                Err(error) => {
+                    tracing::error!(
+                        "could not get database connection while trying to update oracle's answer tx hash: {:#}",
+                        error
+                    );
+                    return Ok(());
+                }
+            };
+
+            if let Err(error) =
+                active_oracle.update_answer_tx_hash(&mut db_connection, tx.tx_hash())
+            {
+                tracing::error!("{:#}", error);
+                return Ok(());
+            }
+        }
+
+        let debug_tx = format!("{:?}", tx);
+        let receipt = match tx.await {
+            Ok(receipt) => receipt,
+            Err(error) => {
+                // we need to throw the following errors as these needs to be addressed immediately.
+                // not being able to delete the tx hash for an oracle once an answer task errors out
+                // might cause a deadlock preventing any answering task from starting in the future
+
+                tracing::error!(
+                    "error while confirming answer transaction {}: {:#}",
```

Comment:
Here too, we could add the oracle address in the error log.

Reply:
Ditto